Drivers/IWDG_driver/Inc/IndependentWatchdog.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * IndependentWatchdog.h
+ *
+ *  Created on: Jun. 22, 2023
+ *      Author: pratyushmakkar
+ */
+
+#ifndef SRC_DRIVERS_INDEPENDENTWATCHDOG_H_
+#define SRC_DRIVERS_INDEPENDENTWATCHDOG_H_
+
+
+#include <stdint.h>
+#include "main.h"
+
+class IWDG_t {
+	private:
+		uint8_t iwdg_prescaler;
+		uint16_t iwdg_reload;
+
+	public:
+		IWDG_t(IWDG_HandleTypeDef* watchdog);
+		IWDG_t(uint8_t iwdg_prescaler, uint16_t iwdg_reload);
```

Comment:
What's the purpose of having the two different constructors?

Reply:
I have handled this to only be a single constructor, where the prescaler and reload can be modified later.